src/main/java/net/unicon/lti/controller/lti/DeepLinkController.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2021 Unicon (R)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.unicon.lti.controller.lti;
+
+import com.google.common.hash.Hashing;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import net.unicon.lti.exceptions.ConnectionException;
+import net.unicon.lti.model.lti.dto.DeepLinkJWTDTO;
+import net.unicon.lti.model.lti.dto.DeepLinkRequest;
+import net.unicon.lti.model.lti.dto.NonceState;
+import net.unicon.lti.service.lti.DeepLinkService;
+import net.unicon.lti.service.lti.LTIJWTService;
+import net.unicon.lti.service.lti.NonceStateService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * This Deep Link controller will return the signed JWT with the deeplinks selected in
+ * the deep link dialog.
+ */
+@Controller
+@Scope("session")
+@RequestMapping("/deeplink")
+public class DeepLinkController {
+
+    static final Logger log = LoggerFactory.getLogger(DeepLinkController.class);
```

Comment:
Would we want to use the `@Log4j` annotation instead?

Reply:
Well, I'm just using the same that is used on all the other code...  If I change this here I will need to change all the logs around the app. That should be another PR.